garage/logger/outputs.py

```diff
@@ -0,0 +1,121 @@
+"""Contains the output classes for the logger.
+
+Each class is sent logger data and handles it itself.
+"""
+import abc
+import csv
+import datetime
+import os
+import sys
+
+import dateutil.tz
+
+from garage.logger.singleton_tabular import TabularInput
+from garage.misc.console import colorize, mkdir_p
+
+
+class LogOutput(abc.ABC):
+    """Abstract class for Logger Outputs."""
+
+    @property
+    def types_accepted(self):
+        """The types that the logger may pass to this output."""
+        return ()
+
+    @abc.abstractmethod
+    def log_output(self, data, **kwargs):
+        """This method is called by the logger when it needs to pass data."""
+        pass
+
+    def dump(self, step=None):
+        """This method is called by the logger to dump an output to file."""
+        pass
+
+
+class StdOutput(LogOutput):
+    """Standard console output for the logger."""
+
+    @property
+    def types_accepted(self):
+        """The types that the logger may pass to this output."""
+        return str, TabularInput
+
+    def log_output(self,
```

Comment:
I'm concerned by this API.

It seems that you are silently accepting args from the upstream `Logger.log()` calls?

@naeioi @krzentner  PTAL

Reply:
Better remove `**kwargs` so that `log_output` rejects unidentified args.

Reply:
should users even be able to propagate arbitrary args/kwargs from `Logger.log()` to `LogOutput.log_output()`?

Reply:
The only use right now for this is so that we can log histograms to `TensorBoardOutput`. If we added the `Histogram` class like @naeioi mentioned we could get rid of this.

Reply:
Maybe that's the easiest solution?